solutions/bronze/usaco-965.mdx

```diff
@@ -180,6 +180,97 @@ public class Lineup {
 </LanguageSection>
 </Spoiler>
 
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$<Asterisk>Note that we don't have to include the time it takes to build permutations and iterate through them because they're constant factors.</Asterisk>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+const int RESTRICT_LEN = 6;
+// list of cows, in alphabetical order
+const string COWS[8] = {
+		"Beatrice",
+		"Belinda",
+		"Bella",
+		"Bessie",
+		"Betsy",
+		"Blue",
+		"Buttercup",
+		"Sue"
+};
+
+vector<vector<string>> orderings;
+void build(vector<string> ordering) {
+	// finished building permutation
+	if ((int)(ordering.size()) == 8) {
+		orderings.push_back(ordering);
+		return;
+	}
+
+	for (string COW : COWS) {
+		if (find(ordering.begin(), ordering.end(), COW) == ordering.end()) {
+			ordering.push_back(COW);
+			build(ordering);
+			ordering.pop_back();
+		}
+	}
+}
+
+// returns index of a cow in an ordering
+int loc(vector<string> &order, string cow) {
+	return find(order.begin(), order.end(), cow) - order.begin();
+}
+
+int main() {
+	freopen("lineup.in", "r", stdin);
+	freopen("lineup.out", "w", stdout);
+	int n;
+	cin >> n;
+
+	vector<pair<string, string>> restrictions;
+
+	for (int i = 0; i < n; i++) {
+
+		string cow1 = "", cow2 = "";
+		for (int j = 0; j < RESTRICT_LEN; j++) {
+			string word;
+			cin >> word;
+
+			if (cow1 == "")
+				cow1 = word;
+			cow2 = word;
+		}
+
+		restrictions.emplace_back(cow1, cow2);
+	}
+
+	build({});
```

Comment:
what's this

Reply:
call the function using an empty vector

Reply:
no crap sherlock
what does calling the function do

Reply:
build the diff orderings, i didn't wanna make the function name too long

should be pretty clear >:[
but i can add a comment if you'd like

Reply:
yeah